Bixal/move.mil
# Jump Links

This PR puts the finishing touches on jump link implementation [see Pivotal ticket](https://www.pivotaltracker.com/story/show/156622661) and fully executes jump link theming [see Pivotal ticket](https://www.pivotaltracker.com/story/show/156622806). With the following caveat.....

**NOTE:** the jump links module/block is currently running on every page of the site, but we really want it restricted to the pages where the jump links appear. We should probably huddle and figure out the most efficient way to do that.

## Checklist

I have…

      against my changes

## Summary of Changes

This pull request…

- Moves the block containing jump links from Drupal's First Sidebar region into the Content section, beneath the page title _(this was the main need)_ and right between the admin links and the Main Content block.
- Used css to place the links and main content in a grid for screen sizes <800px, matching behavior on the live site. The grid is conditional upon existence of the jump links block, so currently it appears on EVERY page, which we don't want to happen. So it's critical that we handle resolution of the issue I pointed out in the note above (only show the block where it's needed) sooner rather than later.
- Added theming for the jump links list itself, exactly matching live site behavior when in a sidebar and, in consultation with Phil, adding a couple small changes on mobile screens for UX purposes.

## Testing

To verify the changes proposed in this pull request…

1. Pull code
1. Update the bixal/jumplinks module (hoping the change to the composer.lock file will allow `make composer` to cover it). 
1. import config and build theme assets
1. Check out the jumplinks on various pages locally, resizing browser to test both mobile and desktop versions.

## Screenshots

### Live site:

[Desktop example](https://www.move.mil/moving-guide/conus):
<img width="288" alt="screen shot 2018-04-18 at 12 57 26 pm" src="https://user-images.githubusercontent.com/29130580/38946667-2e279b20-4308-11e8-9350-039df2243f52.png">

[Mobile example](https://www.move.mil/moving-guide/conus):
<img width="467" alt="screen shot 2018-04-18 at 12 58 47 pm" src="https://user-images.githubusercontent.com/29130580/38946703-46f39d66-4308-11e8-8245-11d57086a92a.png">

### Local as implemented:

Desktop: 
<img width="384" alt="screen shot 2018-04-18 at 12 23 49 pm" src="https://user-images.githubusercontent.com/29130580/38946589-f206a500-4307-11e8-8059-0c6ac760368f.png">

Mobile jumplinks with changes (local with annotation):
![mobile jumplinks example](https://user-images.githubusercontent.com/29130580/38946559-d105988e-4307-11e8-9e06-f1de94d8be77.png)
